.github/workflows/cut-release.yml

```diff
@@ -1,46 +1,133 @@
+# Thanks to this blog post: https://data-dive.com/multi-os-deployment-in-cloud-using-pyinstaller-and-github-actions/
+# Thanks to https://github.com/tauri-apps/tauri-action/tree/v0/
 name: Cut Release
 
 on:
-  - workflow_dispatch
+  workflow_dispatch:
 
 jobs:
-  setup-js:
+  create-release:
+    permissions:
+      contents: write
     runs-on: ubuntu-latest
+    outputs:
+      release_id: ${{ steps.create-release.outputs.result }}
+    
     steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-      - uses: ./.github/actions/setup-node
+      - uses: actions/checkout@v3
 
-  setup-rust:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-      - uses: ./.github/actions/setup-rust
+      - uses: actions/setup-node@v3
 
-  setup-python:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-      - uses: ./.github/actions/setup-python
+      - name: get version
+        run: echo "PACKAGE_VERSION=$(node -p "require('./package.json').version")" >> $GITHUB_ENV
```

Comment:
You could keep these jobs so they run in parallel with `create-release` and so that all jobs `build-tauri` jobs can reuse the cache (don't forget to add `needs: [create-release, setup-js, setup-python, setup-rust]`